privacyidea/lib/user.py

```diff
@@ -597,20 +597,12 @@ def get_user_list(param=None, user=None):
     searchDict = {"username": "*"}
     param = param or {}
 
-    # we have to recreate a new searchdict without the realm key
-    # as delete does not work
-    for key in param:
```

Comment:
You must not drop the ``for``-loop. There could be other search parameters like *email* which you are now missing and which are not contained in the ``searchDict``.

Reply:
You are right, i am on it.

Reply:
Fixed the loop, checks for `username` and `user` (in this order) are done after the loop.